shared/src/main/scala/org/atnos/eff/Union.scala

```diff
@@ -3,31 +3,50 @@ package org.atnos.eff
 /**
  * Union represents one effect T[_] embedded in a tree of possible effects R
  *
- * Since the effect tree is represented with the following cases:
- *   - Fx1[T]
- *   - Fx2[T1, T2]
- *   - Fx3[T1, T2, T3]
+ * The effect tree is represented by four possible cases:
+ *   - fx1[T]
+ *   - fx2[T1, T2]
+ *   - fx3[T1, T2, T3]
  *   - FxAppend[L, R]
  *
- * We have the corresponding Union cases. For example
- *   T2 is in the "middle" of Fx3[T1, T2, T3] so creating a Union object for that effect uses Union3M
+ *  The union type has three concrete constructors:
+ *   - UnionAppendL(nested: Union[L]): Union[FxAppend[L, R]]
+ *   - UnionAppendR(nested: Union[R]): Union[FxAppend[L, R]]
+ *   - UnionTagged(valueUnsafe: Any, index: Int): Union[R] (for R in fx1, fx2, fx3...)
+ *  In that respect UnionTagged behaves similarly to a tagged union in C or C++.
+ *
  */
-sealed trait Union[+R, A] {
-  type X = A
```

Comment:
Why remove the existential type parameter and replace it with `Any` in various places? Wouldn't we keep a bit more type safety with this `X` because then the following continuation has to take a `u.X` and not anything else in scope?